Framework/Geometry/test/InstrumentRayTracerTest.h

```diff
@@ -156,9 +157,13 @@ class InstrumentRayTracerTest : public CxxTest::TestSuite {
    */
   void doTestRectangularDetector(std::string message, Instrument_sptr inst,
                                  V3D testDir, int expectX, int expectY) {
-    //    std::cout << message << '\n';
     InstrumentRayTracer tracker(inst);
-    testDir.normalize(); // Force to be unit vector
+    const auto norm = testDir.norm();
+    if (norm == 0.) {
+      TS_ASSERT_THROWS_ANYTHING(tracker.traceFromSample(testDir));
+      return;
+    }
```

Comment:
This feels like it should be an `else`

Reply:
I moved this case to its own test function.